docs/SdkVersioning.md

```diff
@@ -0,0 +1,590 @@
+# SDK Versioning Strategy
+
+> :warning: THIS IS AN EXPERIMENTAL FEATURE :warning:
+
+## Version Schema
+SDK Versioning (SdkVer) is compatible with SemVer, but not following the full SemVer specification;
+
+```js
+<API>.<RELEASE>.<FIX>(-<PRERELEASE>)
+```
+
+| Item       | Description                                                  |
+| ---------- | ------------------------------------------------------------ |
+| `API`      | Version of the (public) API                                  |
+| `RELEASE`  | Release number                                               |
+| `FIX`      | Incremental number indicating incremental fixes on a release |
+
+In addition, `PRERELEASE` can follow the following two release stages:
+- [Development Releases](#development-releases): `-dev[N]`
+- [Release Candidates](#release-candidates): `-rc[N]`
+
+## Releases
+
+There are three types of Releases supported by SdkVer:
+
+- [Development Releases](#development-releases)
+- [Release Candidates](#release-candidates)
+- [Releases](#releases)
+
+### Development Releases
+
+A Development Release reflects the current state of the `main` development branch.
+The only representation of a Development Release is by the availability of a single
+"draft" GitHub Release, which is updated with every single commit on `main`.
+
+This automatically implies that **no** git tags will be created for each Development Release.
+
+The below table indicates the behavior when creating a Development Release:
+
+| Current Version | Increment Type      | Next Version |
+| --------------- | ------------------- | ------------ |
+| `0.1.0`         | Non-breaking change | `0.2.0-dev1` |
+| `0.2.0-dev1`    | Non-breaking change | `0.2.0-dev2` |
+| `0.1.0`         | Breaking Change     | `1.0.0-dev1` |
+| `0.2.0-dev1`    | Breaking change     | `1.0.0-dev2` |
+
+
+### Release Candidates
+
+It is possible to promote a Development Release to a Release Candidate. This should be used 
+in case:
+- Stabilization is needed before you want to release this version.
+- Development on the `main` branch continued with changes not desired in the release
+
+Each Release Candidate is represented by:
+- a GitHub Release (pre-release state), and
+- an associated git tag on the specified commit.
+
+A Release Candidate is initiated on the `main` development branch;
+
+| Current Version | Increment Type      | Next Version |
+| --------------- | ------------------- | ------------ |
+| `0.1.0-dev1`    | Release Candidate   | `0.1.0-rc1`  |
+| `0.1.0`         | Release Candidate   | `0.2.0-rc1`  |
+
+Stabilizing a Release Candidate is handled on a release branch (i.e. `release/0.1.0`);
+
+| Current Version | Increment Type      | Next Version |
+| --------------- | ------------------- | ------------ |
+| `0.1.0-rc1`     | Non-breaking change | `0.1.0-rc2`  |
+| `0.1.0-rc1`     | Breaking change     | N/A          |
```

Comment:
This is something we discussed as something which might still happen during stabilization of a major release that the API might still be fine-tuned which effectively is a breaking change cross rc1 and rc2. Should we reflect that here in the documentation? Basically as scenario `| 1.0.0-rc1 | Breaking change | 1.0.0-rc2 |`

Reply:
Updated document (after validating against latest status) to be according to your expectations